control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -1318,9 +1316,11 @@ func (r *HostedControlPlaneReconciler) reconcile(ctx context.Context, hostedCont
 	}
 
 	// Reconcile image registry operator
-	r.Log.Info("Reconciling Image Registry Operator")
-	if err := r.reconcileImageRegistryOperator(ctx, hostedControlPlane, releaseImageProvider, userReleaseImageProvider, createOrUpdate); err != nil {
-		return fmt.Errorf("failed to reconcile image registry operator: %w", err)
+	if capabilities.IsImageRegistryCapabilityEnabled(hostedControlPlane.Spec.DisabledCapabilities) {
```

Comment:
We are not plumbing the featuregate flags into the cpo/hcco binaries. You'd need to do it similar to what we do for the HO
https://github.com/openshift/hypershift/blob/main/cmd/install/assets/hypershift_operator.go#L428-L430
https://github.com/openshift/hypershift/blob/main/hypershift-operator/main.go#L156

Or alternative you can just gate on the field being set since it will only exist in the CRD if the HO is installed with --tech-preview-no-upgrade. See autoNode for an example of this.

Reply:
If going on the route of adding a `--feature-gates` flag to cpo and hcco binaries, where/what would set this flag?

While the autoNode approach seems easier, I personally prefer the more explicit one with the feature gate check. Do you have a preference?

Reply:
You'd need to set them similarly to the links above but where we run those binaries, see links below.
https://github.com/openshift/hypershift/blob/main/hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go#L2532
https://github.com/openshift/hypershift/blob/main/control-plane-operator/controllers/hostedcontrolplane/configoperator/reconcile.go#L391

My preference would be to use the api field which is the authoritative source. That way, once you drop the feature gate, the field will be part of the CRD and the business logic will just work with no more changes required. I see value for the flags only for features that are not api driven.